json_response_test.go

```diff
@@ -303,3 +302,42 @@ func TestReadResponseNil(t *testing.T) {
 		})
 	}
 }
+
+func TestReadError(t *testing.T) {
+	type TestStruct struct {
+		Value string
+	}
+
+	tests := []struct {
+		name        string
+		r           io.Reader
+		wantErr     bool
+		wantMessage string
+		wantCode    int
+	}{
+		{"Empty", bytes.NewReader(nil), false, "", 0},
+		{"Response", getResponseBody(TestStruct{"blah"}), false, "", 0},
+		{"Error", getErrorBody("blah", http.StatusNotFound), true, "blah", http.StatusNotFound},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := ReadError(tt.r)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ReadError() error = %v, wantErr %v", err, tt.wantErr)
+			}
+
+			if err != nil {
+				err, ok := err.(*Error)
+				if !ok {
+					t.Fatal("invalid error type")
+				}
+				if got, want := err.Message, tt.wantMessage; got != want {
```

Comment:
Are we just assigning `want := tt.wantMessage` for the sake of readability in `got != want`?

Reply:
Yes, this makes the semantics explicit, and reduces duplication of the struct members in the corresponding `Errorf` below.